src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.removeDeadAnimals();
```

Comment:
[CODE PERFORMANCE] - It is preferred to filter animals with health less than or equal to 0 from `Animal.alive` array using Array's `filter` method. Instead of calling `Animal.removeDeadAnimals()` method inside the `if` statement when an animal's health reaches 0 or less, you should filter the `Animal.alive` array right after decreasing the health of the target. This would avoid the need for an additional check and would be more efficient.